### PR TITLE
feat(governance): implement EPIC I-4 Phase B-1 Hard Gating enforcement

### PR DESCRIPTION
**Pull request by RC918**
# feat(governance): implement EPIC I-4 Phase B-1 Hard Gating enforcement

## Summary

This PR implements the enforcement layer for auto-degradation (EPIC I-4 Phase B-1) that actually applies degradation recommendations to routing decisions. When `DEGRADATION_ENFORCEMENT_ENABLED=true`, providers with AVOID severity are filtered from the available provider list.

**Key changes:**
- Added `DEGRADATION_ENFORCEMENT_ENABLED` feature flag (default: `false`) to settings and env.schema.yaml
- Modified `_get_available_providers()` to filter AVOID providers when enforcement is enabled
- Implemented absolute floor protection ensuring at least 1 provider always remains available
- Added `bypass_governance` parameter for emergency/diagnostic scenarios
- Added `[I-4-ENFORCEMENT]` ERROR-level governance telemetry logging
- Added 10 unit tests covering Hard Gating logic, floor protection, and fail-open behavior

**Safety features:**
- Fail-open: If DegradationAdvisor is unavailable or throws an exception, all providers remain available
- Floor protection: Never returns an empty provider list - uses effective_floor as fallback
- Default disabled: Requires explicit opt-in via environment variable

## Updates since last revision

Addressed reviewer feedback on floor protection semantics:
- **Changed from OR-based to priority-based floor protection**: Previously both `floor_provider` and `fixed_floor` were protected (OR logic). Now uses single `effective_floor` with consistent priority:
  1. Dynamic floor (from DegradationAdvisor) if available and in providers
  2. Fixed floor (from settings) if in providers  
  3. First provider as emergency fallback
- **Consistent logic throughout**: Both filtering and fallback selection now use the same `effective_floor`
- **Added test**: `test_floor_protection_priority_dynamic_over_fixed` verifies priority behavior
- **Kept ERROR log level**: Per user requirement for governance telemetry (this is a governance red line, not normal operation)

## Review & Testing Checklist for Human

- [ ] **Effective floor calculation** (client.py lines 188-198): Verify the priority logic is correct - dynamic floor should take precedence over fixed floor only when it exists AND is in the providers list
- [ ] **Floor protection consistency**: Confirm that `is_floor_protected = (provider == effective_floor)` at line 216 correctly uses the same effective_floor for both filtering and fallback
- [ ] **Integration with DegradationAdvisor**: The `get_degradation_advisor()` and `get_provider_state()` calls are mocked in tests - verify they work correctly with the real advisor in staging
- [ ] **Edge case: dynamic floor not in providers**: If `dynamic_floor` is returned by advisor but not in the available providers list (e.g., filtered by allowlist), verify fallback to fixed_floor works

**Recommended test plan:**
1. Deploy to staging with `DEGRADATION_ADVISORY_ENABLED=true` and `DEGRADATION_ENFORCEMENT_ENABLED=false` (observe-only)
2. Verify [I-4-ADVISORY] logs appear as expected
3. Enable `DEGRADATION_ENFORCEMENT_ENABLED=true` in staging
4. Simulate a provider reaching AVOID state and verify it gets filtered from routing
5. Verify floor protection activates if all providers would be gated
6. Test with different `floor_provider` vs `fixed_floor` values to confirm priority
7. Observe for 48+ hours before production rollout

### Notes

Blueprint Alignment: Model Governance Framework v2 (4.3) - Auto-Degradation

Link to Devin run: https://app.devin.ai/sessions/899162d287fe43538ba78cadfc6a900e
Requested by: Ryan Chen (@RC918)

### CODE REVIEW

File: handoff/20250928/40_App/orchestrator/llm/client.py [hunk 1]
```diff
@@ -69,59 +69,188 @@
...
+        for provider in providers:
+            state = advisor.get_provider_state(provider)
+
+            # Check if provider should be gated
+            if state == DegradationSeverity.AVOID:
+                # Check floor protection
+                is_floor_protected = (
+                    provider == floor_provider or
+                    provider == fixed_floor
+                )
```
Comment (RC918) [inline handoff/20250928/40_App/orchestrator/llm/client.py]: IMPACT: This creates ambiguous floor protection semantics when `floor_provider` and `fixed_floor` differ, leading to unpredictable behavior during enforcement decisions. It violates the principle of least surprise and could result in multiple providers being retained unnecessarily during hard gating. SUGGESTED FIX: Prioritize one strategy over the other (e.g., use dynamic `floor_provider` unless None/unavailable, then fall back to `fixed_floor`).

File: handoff/20250928/40_App/orchestrator/llm/client.py [hunk 2]
```diff
@@ -69,59 +69,188 @@
...
+                    logger.warning(
+                        f"[I-4-ENFORCEMENT] Provider {provider} has AVOID state but is "
+                        f"floor-protected. Keeping in available list."
+                    )
+                else:
+                    # Not floor protected - gate this provider
+                    gated_providers.append(provider)
+                    logger.error(
+                        f"[I-4-ENFORCEMENT] Hard-gating provider {provider} due to "
+                        f"AVOID state."
```
Comment (RC918) [inline handoff/20250928/40_App/orchestrator/llm/client.py]: IMPACT: Using `ERROR` level for expected operational events (enforcement actions) will pollute error metrics and trigger false alerts in monitoring systems. This breaks semantic logging standards where ERROR indicates system malfunction, not policy enforcement. SUGGESTED FIX: Change log level to `WARNING` and introduce structured event fields like `event=hard_gating_applied`, `action=filtered`, `severity=avoid` for proper observability.

File: handoff/20250928/40_App/orchestrator/llm/client.py [hunk 3]
```diff
@@ -69,59 +69,188 @@
...
+                filtered.append(provider)
+
+        # Absolute floor protection: Never return empty list
+        if not filtered and providers:
+            # All providers were gated - keep the first one as emergency fallback
+            fallback = providers[0]
+            if fixed_floor in providers:
+                fallback = fixed_floor
+            elif floor_provider and floor_provider in providers:
+                fallback = floor_provider
```
Comment (RC918) [inline handoff/20250928/40_App/orchestrator/llm/client.py]: IMPACT: The fallback selection logic contradicts the earlier OR-based floor protection check, creating inconsistent behavior between individual provider gating and final floor protection. This could lead to different providers being selected as fallback than those protected during filtering. SUGGESTED FIX: Use a consistent hierarchy (e.g., always prefer `fixed_floor` if set and available, otherwise use `floor_provider`) throughout the function.

**Review by RC918 — COMMENTED** [review]
## MorningAI Code Review

File: handoff/20250928/40_App/orchestrator/llm/client.py (hunk 1 above)
Comment (RC918) [reply handoff/20250928/40_App/orchestrator/llm/client.py]: **已修正 - Floor Protection 語義一致性**

感謝指出這個問題！已在 commit `f50e1ddd` 中修正：

**修改內容：**
- 改用單一 `effective_floor` 取代 OR-based 雙重保護
- 建立一致的優先級邏輯：
  1. Dynamic floor (from DegradationAdvisor) - 如果存在且在 providers 中
  2. Fixed floor (from settings) - 如果在 providers 中
  3. First provider - 緊急 fallback

**優點：**
- 行為可預測（principle of least surprise）
- Filtering 和 fallback 使用相同邏輯
- 新增 `test_floor_protection_priority_dynamic_over_fixed` 測試驗證優先級
